### ISSUE TITLE
SSR: className mismatch in development mode

### ISSUE THREAD

**Issue by jazblueful** [emotion-js/emotion#1490]
I just switched from styled-components to emotion.

My react app and the module responsible for server-side-rendering in my monorepo are decoupled, meaning in production mode they are built separately using webpack and app is imported into ssr module during run-time (same ssr module is used for all apps).

In development however I include the app's source directly into ssr module for hot module replacement to work for both client and the server at the same time.

Production flow:
1. Build from app/src using app.webpack.config -> app/dist
2. Build from ssr/src using ssr.webpack.config -> ssr/dist
3. Run ssr/dist (imports app from app/dist during runtime)
4. Use app/dist for ssr
5. Serve app/dist to client

Development flow:
1. Build app/src using app.webpack.config -> app/dist
2. Build ssr/src using ssr.webpack.config -> ssr/dist (import app from app/src during build)
3. Run ssr/dist
4. Use built-in app for ssr
5. Serve app/dist to client 

**Current behavior:**

In development environment I get ```"Warning: Prop `className` did not match. Server: "css-4y926w" Client: "css-iyqgnx"```

The className properties match in production mode though and I don't get a warning. Also styled-components handled it correctly in both environments.

I digged a little bit into babel-plugin-emotion package and found class-names are hashed based on module name taken from package.json https://github.com/emotion-js/emotion/blob/db69d2282220d0dce52d03f93bc7567795531230/packages/babel-plugin-emotion/src/utils/get-target-class-name.js#L31
https://github.com/emotion-js/emotion/blob/db69d2282220d0dce52d03f93bc7567795531230/packages/babel-plugin-emotion/src/utils/get-target-class-name.js#L39

Obviously in my case the module names for the app and ssr are different.

**Expected behavior:**

classNames should be consistent even if the source is included by differerent packages.

**Environment information:**

- `react` version: 16.8.6
- `emotion` version: 10.0.17

**Comment by Andarist** [comment]
I believe what you have linked to is about generating 'stable classes'. Those are starting with `e` and you have encounter mismatch on regular `css-*` classes. Could you prepare a repro case for this? Would help a lot in finding a root cause.

**Comment by jazblueful** [comment]
I just spent 2 hours to strip the project down to minimal repo in order to demonstrate this issue. At the end when I reinstalled all node_modules to check if the repo would run after cloning it and installing dependencies from scratch, the issue went away!

Same thing with my original repo - reinstalling dependencies fixed it.

I'm going to close this issue for now, if it pops up again and I'm able to reproduce it, I'll reopen.

Thanks
